### PR TITLE
Added "Why is my app getting killed?"

### PR DESCRIPTION
**Pull request by atereshkov**
## Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/Blackjacx/WWDC/blob/master/CONTRIBUTING.md) file.
- [x] I have reviewed my own notes for typos, grammar and consistency (if you're submitting notes for a new session)

## Description
Why is my app getting killed?

### CODE REVIEW

File: summaries/2020/10078.md [hunk 1]
```diff
@@ -2,6 +2,86 @@
...
+- **Most common reasons why apps can be terminated in the background**
+  - **Crash**
+    - Segmentation fault
+    - Illegal instruction
+    - Asserts and uncaught exceptions
+  - **CPU resource limit**
+    - High sustained CPU load in background
+    - Energy Exception Report
+      - Xcode Organizer
+      - `MXCPUExceptionDiagnostic`
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
      - [MXCPUExceptionDiagnostic](https://developer.apple.com/documentation/metrickit/mxcpuexceptiondiagnostic)
```

File: summaries/2020/10078.md [hunk 2]
```diff
@@ -2,6 +2,86 @@
...
+    - Reports contain call stack points out hotspots in code
+    - Consider moving work into `BGProcessingTask`
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
    - Consider moving work into [BGProcessingTask](https://developer.apple.com/documentation/backgroundtasks/bgprocessingtask)
```

File: summaries/2020/10078.md [hunk 3]
```diff
@@ -2,6 +2,86 @@
...
+  - **Watchdog**
+    - Long hang during key app transitions
+      - `application(_:didFinishLaunchingWithOptions:)`
+      - `applicationDidEnterBackground(_:)`
+      - `applicationWillEnterForeground(_:)`
+    - These transitions have a time limit on the order of 20 seconds
+    - Terminations are disabled in Simulator and in the Debugger
+    - Eliminate deadlocks, infinite loops, synchronous work
+    - Report available via `MXCrashDiagnostic`
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
    - Report available via [MXCrashDiagnostic](https://developer.apple.com/documentation/metrickit/mxcrashdiagnostic)
```

File: summaries/2020/10078.md [hunk 4]
```diff
@@ -2,6 +2,86 @@
...
+    - System freeing up memory for active applications
+    - Reducing jetsame rate
+      - Aim for less than 50MB in background
+      - Upon background flush state to disk, clear out image views, drop caches
+    - Recovering from jetsame
+      - Save state upon entering background
+      - Adopt UIKit State Restoration
+      - User should not realize app was terminated
+  - **Background task timeout**
+    - `UIApplication.beginBackgroundTask(expirationHandler:)`
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
    - [UIApplication.beginBackgroundTask(expirationHandler:)](https://developer.apple.com/documentation/uikit/uiapplication/1623031-beginbackgroundtask)
```

File: summaries/2020/10078.md [hunk 5]
```diff
@@ -2,6 +2,86 @@
...
+    - `UIApplication.endBackgroundTask(_:)`
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
    - [UIApplication.endBackgroundTask(_:)`](https://developer.apple.com/documentation/uikit/uiapplication/1622970-endbackgroundtask)
```

File: summaries/2020/10078.md [hunk 6]
```diff
@@ -2,6 +2,86 @@
...
+    - Failure to end the task explicitly result in termination
+    - Counts exposed via `MXBackgroundExitData`
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
    - Counts exposed via [MXBackgroundExitData](https://developer.apple.com/documentation/metrickit/mxbackgroundexitdata)
```

File: summaries/2020/10078.md [hunk 7]
```diff
@@ -2,6 +2,86 @@
...
+    - Preventable
+      - Use the named variant of the UIKit API (`beginBackgroundTask(withName:expirationHandler:)`)
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
      - Use the named variant of the UIKit API ([beginBackgroundTask(withName:expirationHandler:)](https://developer.apple.com/documentation/uikit/uiapplication/1623051-beginbackgroundtask))
```

File: summaries/2020/10078.md [hunk 8]
```diff
@@ -2,6 +2,86 @@
...
+      - Terminations do not occur in Debugger
+      - New console message in iOS 13.4 (_Background Task DatabaseTransaction was created over 30 seconds ago. In Applications running in the background, this creates a risk of termination. Remember to call `endBackgroundTask` for your task in a timely manner to avoid this._)
+    - Expiration handler
+      - Implement an `expirationHandler`
+      - Call `endBackgroundTask(_:)` inside the handler
+      - Do not begin new work
+      - Do not rely on it exclusively
+      - Add telemetry at the start and end of each expiration handler
+      - Inspect `MXMetricPayload`
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
      - Inspect [MXMetricPayload](https://developer.apple.com/documentation/metrickit/mxmetricpayload)
```

File: summaries/2020/10078.md [hunk 9]
```diff
@@ -2,6 +2,86 @@
...
+    - Check `backgroundTimeRemaining`
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
    - Check [backgroundTimeRemaining](https://developer.apple.com/documentation/uikit/uiapplication/1623029-backgroundtimeremaining)
```

File: summaries/2020/10078.md [hunk 10]
```diff
@@ -2,6 +2,86 @@
...
+      - Only start work if plenty of time remains
+      - Unsafe to begin tasks with < 5 seconds remaining
+    - Avoid leaking `UIBackgroundTaskIdentifier`
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
    - Avoid leaking [UIBackgroundTaskIdentifier](https://developer.apple.com/documentation/uikit/uibackgroundtaskidentifier)
```

File: summaries/2020/10078.md [hunk 11]
```diff
@@ -2,6 +2,86 @@
...
+- **New MetricKit API (iOS 14)**
+  - `MXBackgroundExitData` shows how often these terminations happen providing counts of each termination type
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
  - [MXBackgroundExitData](https://developer.apple.com/documentation/metrickit/mxbackgroundexitdata) shows how often these terminations happen providing counts of each termination type
```

File: summaries/2020/10078.md [hunk 12]
```diff
@@ -2,6 +2,86 @@
...
+    - `cumulativeMemoryResourceLimitExitCount`
+    - `cumulativeMemoryPressureExitCount`
+    - `cumulativeSuspendedWithLockedFileExitCount`
+    - `cumulativeAppWatchdogExitCount`
+    - `cumulativeBackgroundTaskAssertionTimeoutExitCount`
+    - `cumulativeNormalExitCount`
+  - Crash reporting via MetricKit
+    - Diagnostics on a per-device basis
+    - Ability to get crash info programmatically directly from the device
+    - `MXCrashDiagnostic`
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
    - [MXCrashDiagnostic](https://developer.apple.com/documentation/metrickit/mxcrashdiagnostic)
```

File: summaries/2020/10078.md [hunk 13]
```diff
@@ -2,6 +2,86 @@
...
+      - Stack trace
+      - Signal
+      - Exception code
+      - Termination reason
+    - Check out [What's New in MetricKit]() to get detailed information
```
Comment (Blackjacx) [inline summaries/2020/10078.md]:
```suggestion
    - Check out [What's New in MetricKit](https://github.com/Blackjacx/WWDC#whats-new-in-metrickit) to get detailed information
```

**Review by Blackjacx — CHANGES_REQUESTED** [review]
Hey this is a very important topic. Thanks for watching. Aso in my must watch list now but thanks to you I can watch it quickly 🙏

I covered all the API with URLs. I think this is a valuable addition for al people reading this since some of the links contain extended further info. Hope you like it 😊

**Comment by atereshkov** [comment]
Thank you @Blackjacx, I just forgot to add links. Added all of them with back-ticks.

**Review by Blackjacx — APPROVED** [review]
Looks good 👍

**Pull request merged**